include/phasar/DataFlow/IfdsIde/Solver/GenericSolverResults.h

```diff
@@ -0,0 +1,258 @@
+/******************************************************************************
+ * Copyright (c) 2022 Fabian Schiebel.
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of LICENSE.txt.
+ *
+ * Contributors:
+ *     Fabian Schiebel and others
+ *****************************************************************************/
+
+#ifndef PHASAR_PHASARLLVM_DATAFLOWSOLVER_IFDSIDE_SOLVER_GENERICSOLVERRESULTS_H
+#define PHASAR_PHASARLLVM_DATAFLOWSOLVER_IFDSIDE_SOLVER_GENERICSOLVERRESULTS_H
+
+#include "phasar/Domain/BinaryDomain.h"
+#include "phasar/Utils/ByRef.h"
+
+#include "llvm/ADT/STLFunctionalExtras.h"
+#include "llvm/IR/Instruction.h"
+#include "llvm/IR/Value.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include <array>
+#include <cstddef>
+#include <cstring>
+#include <new>
+#include <set>
+#include <type_traits>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace psr {
+
+/// XXX: When upgrading to C++20, create a concept checking valid SolverResults
+/// types
+
+/// A type-erased version of the main functionality of SolverResults.
+/// Can be accepted by consumers that don't need deep access to the internals
+/// (so, the usual ones). As we have now two kinds of solver-results
+/// (SolverResults and IdBasedSolverResults), we need a common way of accessing
+/// them.
+template <typename N, typename D, typename L> class GenericSolverResults final {
+public:
+  using n_t = N;
+  using d_t = D;
+  using l_t = L;
+
+  /// NOTE: Don't provide an operator= that constructs a GenericSolverResults
+  /// from a concrete SolverResults object to prevent issues with object
+  /// lifetime. Probably that can be mitigated with use of std::launder, but for
+  /// now, we don't need this complexity
+
+  template <typename T, typename = std::enable_if_t<!std::is_same_v<
+                            GenericSolverResults, std::decay_t<T>>>>
+  GenericSolverResults(const T &SR) noexcept : VT(&VtableFor<std::decay_t<T>>) {
+    using type = std::decay_t<T>;
+
+    static_assert(std::is_trivially_destructible_v<type>);
+    static_assert(std::is_trivially_copyable_v<type>,
+                  "SolverResults should only be a *view* into the results of "
+                  "an IFDS/IDE solver.");
+    static_assert(alignof(type) <= alignof(void *));
```

Comment:
Please add a comment why this is important here.

Reply:
done